waiting/__init__.py

```diff
@@ -13,8 +13,9 @@
 
 def wait(*args, **kwargs):
     result = _Result()
-    for x in iterwait(result=result, *args, **kwargs):
-        pass
+    for exc in iterwait(result=result, *args, **kwargs):
+        if exc:
+            raise exc
```

Comment:
This doesn't take care of the original traceback as raised from the predicate...